atc/creds/retryable_variables_factory.go

```diff
@@ -0,0 +1,65 @@
+package creds
+
+import (
+	"fmt"
+	"github.com/cloudfoundry/bosh-cli/director/template"
+	"github.com/concourse/retryhttp"
+	"time"
+)
+
+type SecretRetryConfig struct {
+	Attempts int           `long:"secret-retry-attempts" default:"5"  description:"The number of attempts secret will be retried to be fetched, in case a retryable error happens."`
+	Interval time.Duration `long:"secret-retry-interval" default:"1s" description:"The interval between secret retry retrieval attempts."`
```

Comment:
Just noticed this is a fixed retry interval - what do you think about using exponential backoff, similar to the Vault manager's login backoff?:

https://github.com/concourse/concourse/blob/78e3a8912f2c1ff413633bf6b22e6d29e95c6acd/atc/creds/vault/reauther.go#L83-L91

Theoretically this would be a bit nicer if the credential manager is failing to handle a deluge of secret requests.

What's here is fine too (better than nothing for sure). Happy to merge this for now and leave exponential backoff for a later PR if you want to investigate it some more.